### PR TITLE
Rename GELU to Gelu

### PR DESCRIPTION
**Pull request by antimora**
This is to follow Rust naming convention. It has been inconsistent how other structs are named.

### Checklist

- [x] Confirmed that `run-checks all` script has been executed.
- [x] Made sure the book is up to date with changes in this PR.

### Changes

1. Rename GELU to Gelu in code
2. Change in docs

### Testing

Ran all checks

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,7 +44,7 @@ Using Burn means having your models optimized on any backend. When possible, we
 automatically and dynamically create custom kernels that minimize data relocation between different
 memory spaces, extremely useful when moving memory is the bottleneck.
 
-As an example, you could write your own GELU activation function with the high level tensor api (see
+As an example, you could write your own Gelu activation function with the high level tensor api (see
```
Comment (louisfd) [inline README.md:47]: I would certainly not change it here because it's not the Rust struct but the deep learning concept

**Review by louisfd — CHANGES_REQUESTED** [review]
I agree there is inconsistency, for instance we have `Lstm` and `Gru` instead of LSTM and GRU, but we also have `ReLU`, `MSELoss`, ... I think if we want to make that move we should rename them too.

**Review by nathanielsimard — APPROVED** [review]

**Comment by antimora** [comment]
@louisfd, it's a good suggestion. I will rename others in a new PR.

**Review by louisfd — APPROVED** [review]
Perfect!

**Pull request merged**